tests/util/setup-dev-tests.ts

```diff
@@ -117,18 +152,102 @@ export function describeDevMoonbeam(
       context.ethers = await context.createEthers();
 
       context.createBlock = async <T>(options: BlockCreation = {}) => {
-        let { parentHash, finalize, transactions = [] } = options;
+        let { parentHash, finalize } = options;
+        return {
+          block: await createAndFinalizeBlock(context.polkadotApi, parentHash, finalize),
+        };
+      };
 
-        let txResults = await Promise.all(
-          transactions.map((t) => customWeb3Request(context.web3, "eth_sendRawTransaction", [t]))
-        );
-        const block = await createAndFinalizeBlock(context.polkadotApi, parentHash, finalize);
+      context.createBlockWithEth = async (
+        transactions: string | string[],
+        options?: BlockCreation
+      ): Promise<EthBlockCreationResponse<string | string[]>> => {
+        if (Array.isArray(transactions)) {
+          const result = await Promise.all(
+            transactions.map((t) => customWeb3Request(context.web3, "eth_sendRawTransaction", [t]))
+          );
+          return {
+            result,
+            block: (await context.createBlock()).block,
+          };
+        }
+        const result = await customWeb3Request(context.web3, "eth_sendRawTransaction", [
+          transactions,
+        ]);
+        const block = (await context.createBlock(options)).block;
+        // Adds extra time to avoid empty transaction when querying it
+        await Promise.resolve((resolve) => setTimeout(resolve, 10));
         return {
-          txResults,
+          result,
           block,
         };
       };
 
+      context.createBlockWithExtrinsic = async <ApiType extends ApiTypes>(
+        transactions: SubmittableExtrinsic<ApiType> | SubmittableExtrinsic<ApiType>[],
+        options?: BlockCreation
+      ): Promise<
+        SubBlockCreationResponse<
+          ApiType,
+          SubmittableExtrinsic<ApiType> | SubmittableExtrinsic<ApiType>[]
+        >
+      > => {
+        // This should return a  string, but is a bit complex to handle type
+        // properly so any will suffice
+        const extrinsicHashes: string[] = [];
+        const txs = Array.isArray(transactions) ? transactions : [transactions];
+        for (const call of txs) {
```

Comment:
I would suggest we use a `for await ... of` loop here to simplify the double `await`.

```typescript
for await (const call of txs) {
  ...
}
```

By accepting `Promise<Call> | Promise<Call>[]` where `Call extends SubmittableExtrinsic<ApiType>>` (transactions: `Promise<SubmittableExtrinsic<ApiType>> | Promise<SubmittableExtrinsic<ApiType>>[]`)

the following

```typescript
await context.createBlockWithExtrinsic(
    [
      await context.polkadotApi.tx.parachainStaking
        .delegate(alith.address, DELEGATION, 0, 0)
        .signAsync(ethan),
      await context.polkadotApi.tx.parachainStaking
        .delegate(alith.address, DELEGATION, 0, 0)
        .signAsync(ethan),
    ]
  );
```

can be simplified to 

```typescript
await context.createBlockWithExtrinsic(
    [
      context.polkadotApi.tx.parachainStaking
        .delegate(alith.address, DELEGATION, 0, 0)
        .signAsync(ethan),
      context.polkadotApi.tx.parachainStaking
        .delegate(alith.address, DELEGATION, 0, 0)
        .signAsync(ethan),
    ]
  );
```

effectively removing all instances of multiple awaits across the tests.

Reply:
Yes, I wanted to add it in a follow-up PR